clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py

```diff
@@ -11,6 +11,10 @@ def populate_result_table_with_list():
     # Populate result table from list with arbitrary table size
     return NotImplementedError()
 
+def append_datasets_in_dict_to_list(recipe_dict, list_to_append):
+    # Append datasets' names from recipe's input or output dictionaries to an input or output list respectively
+    for key in recipe_dict:
+        list_to_append += [x["ref"] for x in recipe_dict[key]["items"]]
```

Comment:
as suggested below, could avoid duplicates here directly 
```suggestion
        list_to_append += [x["ref"] for x in recipe_dict[key]["items"] if x["ref] not in list_to_append]
```

Reply:
Done in commit [b38f5b7ed33c6236e999ede490c40277372a8681](https://github.com/gbetegon88/dataiku-contrib/commit/b38f5b7ed33c6236e999ede490c40277372a8681)